pyproject.toml

```diff
@@ -1,7 +1,63 @@
 [build-system]
-requires = ["setuptools", "lsst-versions"]
+requires = ["setuptools", "lsst-versions >= 1.3.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "felis"
+description = "A vocabulary for describing catalogs and acting on those descriptions"
+license = {text = "GNU General Public License v3 or later (GPLv3+)"}
+readme = "README.md"
+authors = [
+    {name="Rubin Observatory Data Management", email="dm-admin@lists.lsst.org"},
+]
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering :: Astronomy"
+]
+keywords = ["lsst"]
+dependencies = [
+    "astropy >= 4",
+    "sqlalchemy >= 1.4",
+    "click >= 7",
+    "pyyaml >= 6",
+    "pyld >= 2",
+    "pydantic >= 2, < 3"
```

Comment:
I thought we were going to remove this from this PR? This should be on #25 I think.

Reply:
"This" meaning just the line about `pydantic`, @timj ?

Reply:
Yes. Following up on my comment on the draft where we agreed pydantic should not be mentioned here.

Reply:
This will be merged after the Pydantic PR. As `pyproject.toml` doesn't include the dependencies on the Pydantic branch, it makes sense to put it here instead.

Reply:
Ideally you would add it to the setup.cfg file on the other PR and then this PR would move it to the new place. But if you are merging them both today it doesn't really matter.

Reply:
> Ideally you would add it to the setup.cfg file on the other PR and then this PR would move it to the new place. But if you are merging them both today it doesn't really matter.

I didn't change `setup.cfg` on DM-41688, though maybe it should have been updated with the new dependency. It should be resolved once this PR is merged, because that file is changed on this PR to only contain flake8 config and no dependencies.

Reply:
Exactly, it should be on the other and then this should only be moving it, not adding a new dependency at this point. But since they are both merging today it doesn't matter.